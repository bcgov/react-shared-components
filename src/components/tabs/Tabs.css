.bcgov-tab-title {
  border-bottom: none;
}

.bcgov-tab-title a {
  color: #1a5a96 !important;
  background-color: transparent !important;
  text-decoration: underline !important;
  margin-right: 5px;
  margin-left: 10px;
  padding: 0.3em;
  outline: none;
}

.bcgov-tab-title a:not(.active):hover {
  border: 1px solid transparent !important;
}

.bcgov-tab-title a.active {
  border: 1px solid #ddd !important;
  text-decoration: none !important;
  color: #494949 !important;
}

.bcgov-tab-title a.disabled {
  text-decoration: none !important;
  color: lightgray !important;
}

.bcgov-tab-content {
  border: 1px solid #ddd;
  border-radius: 0.2em;
  padding: 0.8em;
  overflow: auto;
}
