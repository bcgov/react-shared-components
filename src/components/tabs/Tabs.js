import React from "react";
import BootTabs from "react-bootstrap/Tabs";
import BootTab from "react-bootstrap/Tab";
import PropTypes from "prop-types";

export const Tabs = ({ tabs, styling, height }) => {
  return (
    <div className={`bcgov-tabs ${styling}`}>
      <BootTabs
        className="bcgov-tab-title"
        id="uncontrolled-tab-example"
        defaultActiveKey={tabs[0].title}
        transition={false}
      >
        {tabs.map((tab) => (
          <BootTab
            className="bcgov-tab-content"
            eventKey={tab.title}
            key={tab.title}
            title={tab.title}
            disabled={tab.isDisabled}
            style={{ height: `${height}px` }}
          >
            {tab.content}
          </BootTab>
        ))}
      </BootTabs>
    </div>
  );
};

Tabs.propTypes = {
  styling: PropTypes.string,
  height: PropTypes.number.isRequired,
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      content: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
        .isRequired,
      isDisabled: PropTypes.bool,
    })
  ).isRequired,
};

Tabs.defaultProps = {
  styling: "",
};
