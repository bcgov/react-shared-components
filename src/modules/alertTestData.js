import React from "react";
import { Alert } from "../components/alert/Alert";

export function generateAlert(icon, type, styling, element) {
  return <Alert icon={icon} type={type} styling={styling} element={element} />;
}
