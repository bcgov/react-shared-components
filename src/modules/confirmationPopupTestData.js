/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getConfirmationPopupTestData() {
  return { body };
}
