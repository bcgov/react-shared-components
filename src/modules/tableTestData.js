const elements = [
  { name: "BCeID", value: "bobross42" },
  {
    name: "Last Name",
    value: "Ross",
  },
  { name: "First Name", value: "Bob" },
  { name: "Email Address", value: "bob.ross@example.com" },
];

export function getTableElementsTestData() {
  return elements;
}
