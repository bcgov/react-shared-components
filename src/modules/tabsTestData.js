import React from "react";

export const tabs = [
  {
    title: "Tab1",
    content: <p> Tab1 Content... </p>,
  },
  {
    title: "Tab2",
    content: <p> Tab2 Content... </p>,
  },
  {
    title: "Tab3",
    content: <p> Tab3 Content... </p>,
    isDisabled: true,
  },
];
