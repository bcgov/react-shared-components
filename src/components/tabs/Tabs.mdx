import { Meta, Story } from "@storybook/addon-docs/blocks";
import { Tabs } from "./Tabs";

<Meta title="Tabs" component={Tabs} />

# Tabs

The tabs component is used to display information in an easy tab navigator.

## API

| Prop    | Type             | Required | Description                                               | Default |
| ------- | ---------------- | -------- | --------------------------------------------------------- | ------- |
| styling | string           | false    | css classnames representing the styling you wish to apply | ""      |
| tabs    | array of objects | true     | title and content of the tabs                             |         |
| height  | number           | true     | height of tabs element in pixels                          |         |

## Tabs API

The tabs array contains objects of the following structure:

| Prop       | Type           | Required | Description                            | Default |
| ---------- | -------------- | -------- | -------------------------------------- | ------- |
| title      | string         | true     | the title of a tab                     |         |
| content    | string/element | true     | the content of a tab                   |         |
| isDisabled | boolean        | false    | represents if a tab should be disabled |         |

<br />

## Tabs

<Story name="BC Gov Tabs" id="tabs" height={400}></Story>

```js
const tabs = [
  {
    title: "Tab1",
    content: <p> Tab Content... </p>,
  },
  {
    title: "Tab2",
    content: <p> Tab Content... </p>,
  },
  {
    title: "Tab3",
    content: <p> Tab Content... </p>,
    isDisabled: true,
  },
];

<Tabs tabs={tabs} height={300} />;
```
