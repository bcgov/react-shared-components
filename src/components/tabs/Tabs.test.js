import React from "react";

import testBasicSnapshot from "../../TestHelper";
import { Tabs } from "./Tabs";
import { tabs } from "../../modules/tabsTestData";

function testTabs() {
  testBasicSnapshot(<Tabs tabs={tabs} height={300} />);
}

describe("Tabs Component", () => {
  test("Matches the snapshot", () => {
    testTabs();
  });
});
