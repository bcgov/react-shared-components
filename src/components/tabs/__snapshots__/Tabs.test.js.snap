// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component Matches the snapshot 1`] = `
<div
  className="bcgov-tabs "
>
  <nav
    className="bcgov-tab-title nav nav-tabs"
    onKeyDown={[Function]}
    role="tablist"
  >
    <a
      aria-controls="uncontrolled-tab-example-tabpane-Tab1"
      aria-selected={true}
      className="nav-item nav-link active"
      data-rb-event-key="Tab1"
      href="#"
      id="uncontrolled-tab-example-tab-Tab1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      Tab1
    </a>
    <a
      aria-controls="uncontrolled-tab-example-tabpane-Tab2"
      aria-selected={false}
      className="nav-item nav-link"
      data-rb-event-key="Tab2"
      href="#"
      id="uncontrolled-tab-example-tab-Tab2"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Tab2
    </a>
    <a
      aria-controls="uncontrolled-tab-example-tabpane-Tab3"
      aria-disabled={true}
      aria-selected={false}
      className="nav-item nav-link disabled"
      data-rb-event-key="Tab3"
      href="#"
      id="uncontrolled-tab-example-tab-Tab3"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Tab3
    </a>
  </nav>
  <div
    className="tab-content"
  >
    <div
      aria-hidden={false}
      aria-labelledby="uncontrolled-tab-example-tab-Tab1"
      className="bcgov-tab-content tab-pane active"
      id="uncontrolled-tab-example-tabpane-Tab1"
      role="tabpanel"
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <p>
         Tab Content... 
      </p>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="uncontrolled-tab-example-tab-Tab2"
      className="bcgov-tab-content tab-pane"
      id="uncontrolled-tab-example-tabpane-Tab2"
      role="tabpanel"
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <p>
         Tab Content... 
      </p>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="uncontrolled-tab-example-tab-Tab3"
      className="bcgov-tab-content tab-pane"
      id="uncontrolled-tab-example-tabpane-Tab3"
      role="tabpanel"
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <p>
         Tab Content... 
      </p>
    </div>
  </div>
</div>
`;

exports[`Tabs Component Matches the snapshot 2`] = `
<div
  className="bcgov-tabs "
>
  <nav
    className="bcgov-tab-title nav nav-tabs"
    onKeyDown={[Function]}
    role="tablist"
  >
    <a
      aria-controls="uncontrolled-tab-example-tabpane-Tab1"
      aria-selected={true}
      className="nav-item nav-link active"
      data-rb-event-key="Tab1"
      href="#"
      id="uncontrolled-tab-example-tab-Tab1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      Tab1
    </a>
    <a
      aria-controls="uncontrolled-tab-example-tabpane-Tab2"
      aria-selected={false}
      className="nav-item nav-link"
      data-rb-event-key="Tab2"
      href="#"
      id="uncontrolled-tab-example-tab-Tab2"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Tab2
    </a>
    <a
      aria-controls="uncontrolled-tab-example-tabpane-Tab3"
      aria-disabled={true}
      aria-selected={false}
      className="nav-item nav-link disabled"
      data-rb-event-key="Tab3"
      href="#"
      id="uncontrolled-tab-example-tab-Tab3"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
      tabIndex={-1}
    >
      Tab3
    </a>
  </nav>
  <div
    className="tab-content"
  >
    <div
      aria-hidden={false}
      aria-labelledby="uncontrolled-tab-example-tab-Tab1"
      className="bcgov-tab-content tab-pane active"
      id="uncontrolled-tab-example-tabpane-Tab1"
      role="tabpanel"
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <p>
         Tab Content... 
      </p>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="uncontrolled-tab-example-tab-Tab2"
      className="bcgov-tab-content tab-pane"
      id="uncontrolled-tab-example-tabpane-Tab2"
      role="tabpanel"
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <p>
         Tab Content... 
      </p>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="uncontrolled-tab-example-tab-Tab3"
      className="bcgov-tab-content tab-pane"
      id="uncontrolled-tab-example-tabpane-Tab3"
      role="tabpanel"
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <p>
         Tab Content... 
      </p>
    </div>
  </div>
</div>
`;
